Lib/fontTools/varLib/builder.py

```diff
@@ -68,6 +68,16 @@ def buildVarData(varRegionIndices, items, optimize=True):
 			assert len(item) == regionCount
 			records.append(list(item))
 	self.ItemCount = len(self.Item)
+	numShorts = 0
+	for item in records:
+		assert len(item) == regionCount, ("Item length mismatch", len(item), regionCount)
+		for i in range(regionCount - 1, numShorts - 1, -1):
+			if not (-128 <= item[i] <= 127):
+				numShorts = i + 1
+				break
+		if numShorts == regionCount:
+			break
+	self.NumShorts = numShorts
```

Comment:
This is wrong.  This block should go *after* call to optimizeVarData(). In fact, maybe move code from that function here, and only make the loop that calls _reorderItem() be dependent on "if optimize".

Rest looks great. Thanks!